### PR TITLE
Upgrade from Chromium 92 to Chromium 93

### PR DESCRIPTION
**Pull request by mkarolin**
Fixes brave/brave-browser#16306
Related PR: brave/brave-browser#16679

## Submitter Checklist:

- [ ] I confirm that no security/privacy review [is needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or that I have [requested](https://github.com/brave/security/issues/new/choose) one
- [ ] There is a [ticket](https://github.com/brave/brave-browser/issues) for my issue
- [ ] Used Github [auto-closing keywords](https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue) in the PR description above
- [ ] Wrote a good [PR/commit description](https://google.github.io/eng-practices/review/developer/cl-descriptions.html)
- [ ] Added appropriate labels (`QA/Yes` or `QA/No`; `release-notes/include` or `release-notes/exclude`; `OS/...`) to the associated issue
- [ ] Checked the PR locally: `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests`, `npm run lint`, `npm run gn_check`, `npm run tslint`
- [ ] Ran `git rebase master` (if needed)

## Reviewer Checklist:

- [ ] A security review [is not needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or a link to one is included in the PR description
- [ ] New files have MPL-2.0 license header
- [ ] Adequate test coverage exists to prevent regressions
- [ ] Major classes, functions and non-trivial code blocks are well-commented
- [ ] Changes in component dependencies are properly reflected in `gn`
- [ ] Code follows the [style guide](https://chromium.googlesource.com/chromium/src/+/HEAD/styleguide/c++/c++.md)
- [ ] Test plan is specified in PR before merging

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on
- [ ] All relevant documentation has been updated, for instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protections
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Custom-Headers
  - [ ] https://github.com/brave/brave-browser/wiki/Web-Compatibility-Exceptions-in-Brave
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide
  - [ ] https://github.com/brave/brave-browser/wiki/P3A

### CODE REVIEW

File: components/brave_wallet/browser/rlp_encode.cc [hunk 1]
```diff
@@ -50,8 +50,11 @@ base::Value RLPUint256ToBlobValue(uint256_t input) {
 
 std::string RLPEncode(base::Value val) {
   if (val.is_int()) {
-    int i;
-    val.GetAsInteger(&i);
+    int i = 0;
+    bool is_integer = val.is_int();
```
[the remark this answers is not in the document]
Comment (mkarolin) [reply components/brave_wallet/browser/rlp_encode.cc]: Addressed in 94dcf987ab6772650d31106990439f5d9bd503be

File: components/sync/driver/brave_sync_profile_sync_service.h [hunk 2]
```diff
@@ -21,7 +21,7 @@ namespace syncer {
 class BraveSyncAuthManager;
 class ProfileSyncServiceDelegate;
 
-class BraveProfileSyncService : public ProfileSyncService {
+class BraveProfileSyncService : public SyncServiceImpl {
```
[the remark this answers is not in the document]
Comment (mkarolin) [reply components/sync/driver/brave_sync_profile_sync_service.h]: @darkdh did some renaming in 34a6ca2914b7103e9d545961a4c4aea5f709de7f and in 65878ba065e634189a79e705942893820aa4b04c

File: patches/chrome-browser-ui-android-omnibox-java-src-org-chromium-chrome-browser-omnibox-suggestions-DropdownItemViewInfoListBuilder.java.patch [hunk 3]
```diff
@@ -0,0 +1,13 @@
+diff --git a/chrome/browser/ui/android/omnibox/java/src/org/chromium/chrome/browser/omnibox/suggestions/DropdownItemViewInfoListBuilder.java b/chrome/browser/ui/android/omnibox/java/src/org/chromium/chrome/browser/omnibox/suggestions/DropdownItemViewInfoListBuilder.java
+index 208580a0d53c8a578de68b0f25dd0f3ecf823e05..4f963f68b0d2ed8178fd399904d6db6ff49bde2f 100644
+--- a/chrome/browser/ui/android/omnibox/java/src/org/chromium/chrome/browser/omnibox/suggestions/DropdownItemViewInfoListBuilder.java
++++ b/chrome/browser/ui/android/omnibox/java/src/org/chromium/chrome/browser/omnibox/suggestions/DropdownItemViewInfoListBuilder.java
+@@ -92,7 +92,7 @@ class DropdownItemViewInfoListBuilder {
+                 () -> mShareDelegateSupplier == null ? null : mShareDelegateSupplier.get();
+ 
+         mHeaderProcessor = new HeaderProcessor(context, host, delegate);
+-        registerSuggestionProcessor(new EditUrlSuggestionProcessor(
++        registerSuggestionProcessor(new new org.chromium.chrome.browser.omnibox.suggestions.editurl.BraveEditUrlSuggestionProcessor(
```
Comment (bridiver) [inline patches/chrome-browser-ui-android-omnibox-java-src-org-chromium-chrome-browser-omnibox-suggestions-DropdownItemViewInfoListBuilder.java.patch]: `new new`
also looks like something we could use asm to do

File: patches/content-public-browser-web_contents_receiver_set.h.patch [hunk 4]
```diff
@@ -0,0 +1,25 @@
+diff --git a/content/public/browser/web_contents_receiver_set.h b/content/public/browser/web_contents_receiver_set.h
+index c373078ea48ebf2886b3a42513eb795d24da159d..8cf8525f3a3b5722b39766d8400854ba78284858 100644
+--- a/content/public/browser/web_contents_receiver_set.h
++++ b/content/public/browser/web_contents_receiver_set.h
+@@ -59,6 +59,11 @@ namespace subresource_filter {
+ class ContentSubresourceFilterThrottleManager;
+ }
+ 
++class BraveDrmTabHelper;
```
Comment (bridiver) [inline patches/content-public-browser-web_contents_receiver_set.h.patch:9]: this is definitely not the right way to patch this, is this temporary or something?

File: patches/chrome-browser-ui-android-omnibox-java-src-org-chromium-chrome-browser-omnibox-suggestions-DropdownItemViewInfoListBuilder.java.patch (hunk 3 above)
Comment (samartnik) [reply patches/chrome-browser-ui-android-omnibox-java-src-org-chromium-chrome-browser-omnibox-suggestions-DropdownItemViewInfoListBuilder.java.patch]: Yes, it was fixed later https://github.com/samartnik/brave-core/commit/d77d5d8a5efb87577903438f4eaecac124272694

File: browser/search/ntp_utils.cc [hunk 5]
```diff
@@ -9,7 +9,7 @@
 #include "brave/browser/profiles/profile_util.h"
 #include "brave/common/pref_names.h"
 #include "brave/components/crypto_dot_com/browser/buildflags/buildflags.h"
-#include "components/ntp_tiles/pref_names.h"
+#include "chrome/browser/ui/webui/new_tab_page/ntp_pref_names.h"
```
Comment (bridiver) [inline browser/search/ntp_utils.cc:12]: is there a dep we can remove for `//components/ntp_tiles`?

File: browser/brave_prefs_browsertest.cc [hunk 6]
```diff
@@ -13,10 +13,10 @@
 #include "chrome/browser/browser_process.h"
 #include "chrome/browser/net/prediction_options.h"
 #include "chrome/browser/profiles/profile.h"
+#include "chrome/browser/ui/webui/new_tab_page/ntp_pref_names.h"
```
Comment (bridiver) [inline browser/brave_prefs_browsertest.cc:16]: is there a dep we can remove for //components/ntp_tiles?

File: build/commands/lib/config.js [hunk 7]
```diff
@@ -409,7 +409,7 @@ Config.prototype.buildArgs = function () {
     // Component builds are not supported for iOS:
     // https://chromium.googlesource.com/chromium/src/+/master/docs/component_build.md
     args.is_component_build = false
-    args.ios_deployment_target = '12.0'
+    args.ios_deployment_target = '13.0'
```
Comment (bridiver) [inline build/commands/lib/config.js:412]: we should remove this line. We don't want to override the default value here

File: ios/browser/api/sync/brave_sync_worker.cc [hunk 8]
```diff
@@ -318,7 +318,7 @@ bool BraveSyncWorker::IsSyncEnabled() {
     return false;
   }
 
-  return setup_service->IsSyncEnabled();
+  return setup_service->CanSyncFeatureStart();
```
Comment (bridiver) [inline ios/browser/api/sync/brave_sync_worker.cc:321]: `BraveSyncWorker::IsSyncEnabled` -> `CanSyncFeatureStart` for consistency

File: patches/chrome-browser-resources-settings-route.js.patch [hunk 9]
```diff
@@ -1,14 +1,21 @@
...
+-
++import addBraveRoutes from './brave_routes.js'
+ /**
+  * Add all of the child routes that originate from the privacy route,
+  * regardless of whether the privacy section under basic or advanced.
+@@ -214,6 +214,7 @@ function createBrowserSettingsRoutes() {
+       // </if>
+     }
+   }
++  addBraveRoutes(r)
```
Comment (bridiver) [inline patches/chrome-browser-resources-settings-route.js.patch:18]: can we import in this same line?

File: browser/search/ntp_utils.cc (hunk 5 above)
Comment (mariospr) [reply browser/search/ntp_utils.cc:12]: The `//components/ntp_tiles` GN target was not explicitly listed for `//brave/browser/search` prior to this change, so there's nothing to remove.

File: browser/brave_prefs_browsertest.cc (hunk 6 above)
Comment (mariospr) [reply browser/brave_prefs_browsertest.cc:16]: The `//components/ntp_tiles` GN target was not explicitly listed for `//brave/test:brave_unit_tests` prior to this change, so there's nothing to remove.

File: patches/chrome-browser-resources-settings-route.js.patch (hunk 9 above)
Comment (mariospr) [reply patches/chrome-browser-resources-settings-route.js.patch:18]: cc @petemill

File: components/decentralized_dns/decentralized_dns_opt_in_page.h [hunk 10]
```diff
@@ -9,6 +9,7 @@
 #include <memory>
 #include <string>
 
+#include "base/values.h"
```
Comment (iefremov) [inline components/decentralized_dns/decentralized_dns_opt_in_page.h:12]: could be a forward decl?

File: patches/content-public-browser-web_contents_receiver_set.h.patch (hunk 4 above)
Comment (mariospr) [reply patches/content-public-browser-web_contents_receiver_set.h.patch:9]: @bridiver You're right on this was meant to be a temporary thing and needs to be fixed. At the time I fixed the issue I read the threads [in the CL](review.googlesource.com/c/chromium/src/+/2956368) about how this was meant to be temporary and there were plans to fix them progressively, but there were no migrations done yet at the time (I wrote this on June 23rd, and the [first migration upstream is from July 14th](https://bugs.chromium.org/p/chromium/issues/detail?id=1213679#c3)).

Then I somehow forgot to check again once I was back from holidays and now it's not only that there are plenty of migrations done upstream, but also that even [WebContentsReceiverSet has been removed](https://bugs.chromium.org/p/chromium/issues/detail?id=1213679#c28)!

I'll work on a fix for this now, thanks for catching this!

File: components/decentralized_dns/decentralized_dns_opt_in_page.h (hunk 10 above)
Comment (mariospr) [reply components/decentralized_dns/decentralized_dns_opt_in_page.h]: Fixed in https://github.com/brave/brave-core/commit/c4fc3b521dd321373d7e46222752410bb9357e0f

File: browser/ui/views/profiles/brave_profile_menu_view.cc [hunk 11]
```diff
@@ -25,7 +25,7 @@
 
 namespace {
 bool IsGuest(Profile* profile) {
```
Comment (iefremov) [inline browser/ui/views/profiles/brave_profile_menu_view.cc:27]: mb we don't need this whole function now?

File: patches/content-public-browser-web_contents_receiver_set.h.patch (hunk 4 above)
Comment (mariospr) [reply patches/content-public-browser-web_contents_receiver_set.h.patch]: Fixed on https://github.com/brave/brave-core/commit/c698321ba83e0fdec507e3c3627c02f70a5afbda and https://github.com/brave/brave-core/commit/57354fecd87bcd8f1936b1a7dca5276009a53a0a

File: browser/ui/views/profiles/brave_profile_menu_view.cc (hunk 11 above)
Comment (mariospr) [reply browser/ui/views/profiles/brave_profile_menu_view.cc]: Correct, fixed in https://github.com/brave/brave-core/commit/5adfdcb18e7949701c005d18070216a13016aac4

**Review by Brandon-T — APPROVED** [review]

File: build/commands/lib/config.js (hunk 7 above)
Comment (mkarolin) [reply build/commands/lib/config.js]: Removed in c7df32f9ab0edf123f6b8f6faad6d1ada8b10882

File: ios/browser/api/sync/brave_sync_worker.cc (hunk 8 above)
Comment (mkarolin) [reply ios/browser/api/sync/brave_sync_worker.cc:321]: Addressed in 70e2a74dc8d246dab6e40d946e90f17be9e42d3c

File: patches/chrome-browser-resources-settings-route.js.patch (hunk 9 above)
Comment (mkarolin) [reply patches/chrome-browser-resources-settings-route.js.patch:18]: Doesn't seem like we can have import on the same line:
<pre>
[!] Error: 'import' and 'export' may only appear at the top level
</pre>

File: patches/third_party-blink-renderer-platform-media-key_system_config_selector.cc.patch [hunk 12]
```diff
@@ -0,0 +1,12 @@
+diff --git a/third_party/blink/renderer/platform/media/key_system_config_selector.cc b/third_party/blink/renderer/platform/media/key_system_config_selector.cc
+index 86d68a45b24824d9b8ce855ffec5646f8b090483..905ea089c6ba6857cf0508affb2a7af404f39c76 100644
+--- a/third_party/blink/renderer/platform/media/key_system_config_selector.cc
++++ b/third_party/blink/renderer/platform/media/key_system_config_selector.cc
+@@ -995,6 +995,7 @@ void KeySystemConfigSelector::SelectConfig(
+ 
+   key_systems_->UpdateIfNeeded();
+ 
++  media::KeySystems::GetInstance();
```
[the remark this answers is not in the document]
Comment (bridiver) [reply patches/third_party-blink-renderer-platform-media-key_system_config_selector.cc.patch:9]: I was wondering about that as well

File: patches/ui-views-controls-focus_ring.cc.patch [hunk 13]
```diff
@@ -0,0 +1,12 @@
+diff --git a/ui/views/controls/focus_ring.cc b/ui/views/controls/focus_ring.cc
+index 0de6ff1ecffaaddd262692dfd949ab9c59ee5e0d..28bfaccde6099e3cc3e716eb5b571a00a3fe6373 100644
+--- a/ui/views/controls/focus_ring.cc
++++ b/ui/views/controls/focus_ring.cc
+@@ -191,6 +191,7 @@ void FocusRing::OnPaint(gfx::Canvas* canvas) {
+   cc::PaintFlags paint;
+   paint.setAntiAlias(true);
+   paint.setColor(color_.value_or(GetColor(this, !invalid_)));
++  BRAVE_FOCUS_RING_ON_PAINT_SET_COLOR_VALIDITY
```
[the remark this answers is not in the document]
Comment (mkarolin) [reply patches/ui-views-controls-focus_ring.cc.patch]: 👍 Implemented in 938bc40bf08da91ca6066c969711a07196e2cd43

**Review by darkdh — APPROVED** [review]

**Review by bridiver — APPROVED** [review]

File: patches/third_party-blink-renderer-platform-media-key_system_config_selector.cc.patch (hunk 12 above)
[the remark this answers is not in the document]
Comment (simonhong) [reply patches/third_party-blink-renderer-platform-media-key_system_config_selector.cc.patch:9]: Yes! we can delete this patch :)

File: chromium_src/chrome/browser/about_flags.cc [hunk 14]
```diff
@@ -3,38 +3,7 @@
...
-#include "chrome/browser/profiles/profile_manager.h"
-#include "components/prefs/pref_service.h"
 
-#define SetFeatureEntryEnabled SetFeatureEntryEnabled_ChromiumImpl
-#include "../../../../chrome/browser/about_flags.cc"  // NOLINT
-#undef SetFeatureEntryEnabled
-
-namespace about_flags {
-
-void UpdateBraveMediaRouterPref(const std::string& internal_name,
```
Comment (goodov) [inline chromium_src/chrome/browser/about_flags.cc:20]: is this delete intentional?
```SetFeatureEntryEnabled``` and ```UpdateBraveMediaRouterPref```

**Review by goodov — APPROVED** [review]

**Review by zenparsing — APPROVED** [review]

**Review by iefremov — APPROVED** [review]

File: patches/third_party-blink-renderer-platform-media-key_system_config_selector.cc.patch (hunk 12 above)
[the remark this answers is not in the document]
Comment (mkarolin) [reply patches/third_party-blink-renderer-platform-media-key_system_config_selector.cc.patch]: Removed in 9d182b392ff7aa1439edf8effbf5cce84b5c4390

**Review by samartnik — APPROVED** [review]
++

**Comment by mkarolin** [comment]
**When uplifting, also uplift:**
* https://github.com/brave/brave-browser/issues/17549

**Pull request merged**